testing/internal/e2e/tests/base/paginate_auth_token_test.go

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package base_test
+
+import (
+	"context"
+	"encoding/json"
+	"slices"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/authmethods"
+	"github.com/hashicorp/boundary/api/authtokens"
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCliPaginateAuthTokens asserts that the CLI automatically paginates to retrieve
+// all auth tokens in a single invocation.
+func TestCliPaginateAuthTokens(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadTestConfig()
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+	client, err := boundary.NewApiClient()
+	require.NoError(t, err)
+	newOrgId := boundary.CreateNewOrgCli(t, ctx)
+	userId := boundary.CreateNewUserApi(t, ctx, client, newOrgId)
+	amId := boundary.CreateNewAuthMethodApi(t, ctx, client, newOrgId)
+	accId, password := boundary.CreateNewAccountCli(t, ctx, amId, "testuser")
+	boundary.SetAccountToUserCli(t, ctx, userId, accId)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("auth-methods", "delete", "-id", amId))
+		require.NoError(t, output.Err, string(output.Stderr))
+		output = e2e.RunCommand(ctx, "boundary", e2e.WithArgs("scopes", "delete", "-id", newOrgId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	boundary.AuthenticateCli(t, ctx, amId, "testuser", password)
+
+	// Create enough auth tokens to overflow a single page.
+	// We created an auth token when we authenticated, so we
+	// need to remove it from the total auth token count.
+	numPrecreatedAuthTokens := 1
```

Comment:
If a number of tests have already executed prior to this test running, wouldn't there be a number of auth tokens (not just 1) already created? The test is passing in CI, but it might have more auth tokens in the list than expected?

EDIT: But that wouldn't make sense since that would've failed this check below...
```
require.Len(t, newAuthTokens.Items, c.MaxPageSize+1)
```

EDIT2: Ah, we're just listing auth tokens in that org, which only have tokens from the login attempts using this new auth method.